Modules/_datetimemodule.c

```diff
@@ -1506,6 +1506,32 @@ format_utcoffset(char *buf, size_t buflen, const char *sep,
     return 0;
 }
 
+static PyObject *
+make_somezreplacement(PyObject *object, char *sep, PyObject *tzinfoarg)
+{
+    char buf[100];
+    PyObject *tzinfo = get_tzinfo_member(object);
+    PyObject *replacement = PyBytes_FromStringAndSize(NULL, 0);
+
+    if (replacement == NULL)
+        return NULL;
+    if (tzinfo == Py_None || tzinfo == NULL)
+        return replacement;
+
+    Py_DECREF(replacement);
+
+    assert(tzinfoarg != NULL);
+    if (format_utcoffset(buf,
+                         sizeof(buf),
+                         sep,
+                         tzinfo,
+                         tzinfoarg) < 0)
+        return NULL;
+
+    replacement = PyBytes_FromStringAndSize(buf, strlen(buf));
+    return replacement;
```

Comment:
```suggestion
make_somezreplacement(PyObject *object, char *sep, PyObject *tzinfoarg)
{
    char buf[100];
    PyObject *tzinfo = get_tzinfo_member(object);

    if (tzinfo == Py_None || tzinfo == NULL) {
        return PyBytes_FromStringAndSize(NULL, 0);
    }
   
    assert(tzinfoarg != NULL);
    if (format_utcoffset(buf,
                         sizeof(buf),
                         sep,
                         tzinfo,
                         tzinfoarg) < 0)
        return NULL;
        
    return PyBytes_FromStringAndSize(buf, strlen(buf));
```

Reply:
This avoids a lot of error handling.

Reply:
Yeah, thanks - I committed your version.